lib/graphql/batch/loader.rb

```diff
@@ -92,6 +95,14 @@ def cache_key(load_key)
 
     private
 
+    def finish_resolve(key)
+      promise = promise_for(key)
+      return yield(promise) unless executor
```

Comment:
is this line necessary since it will raise if not wrapped in `GraphQL::Batch.batch` in `for`?

Reply:
The executor is mostly just used for storing loaders so they can be looked up for grouping, however, an alternative would be to store loaders in a context.  We started supporting using loaders without an executor in https://github.com/Shopify/graphql-batch/pull/35.

Reply:
oh yes, of course! 👍